### DISCUSSION TITLE
Lista w fromacie dnsmasq bez konwertowania jej na obcych serwerach

### DISCUSSION THREAD

**Discussion by krystian3w** [FiltersHeroes/KAD#2389] [Pomysły]
Paweł B. pisał wczoraj na mejla:

> Cześć,
> 
> Zacząłem ostatnio używać routera na OpenWRT oraz wtyczki adblock-lean. Jeden "problem" to wymóg list w formacie **dnsmasq**. Czy jest możliwość, żeby KAD było publikowane także w nim?
> 
> Pozdrawiam,
> Paweł B.

Wstępnie by była to lista domen bez IP (może być ułomne w banowaniu IP, nie używam OpenWRT ani adblock-lean do żadnego routera) w zapisie:

```hosts
# Title: KADdnsmasq
# Description: Filtry chroniace przed roznego rodzaju przekretami i oszustwami, takimi jak np. subskrypcje SMS, wirusy i falszywe sklepy internetowe. Oprocz wlasnego zestawu, zawieraja one rowniez wpisy z list stworzonych przez takie zespoly jak np. CERT Polska, legalniewsieci.pl i KNF.
# Last modified: Mon, 15 Jan 2024, 17:21 UTC+01:00
# Version: 2024.1.15.2
# Expires: 2 days
# Website: https://kadantiscam.netlify.app/
# Based on: https://github.com/FiltersHeroes/KAD
# License: CC BY-SA 4.0 (https://creativecommons.org/licenses/by-sa/4.0/)
# Creator: azet12
# Maintainer: Filters Heroes
#
# (EN) WARNING: Sites listed below can be harmful. Please, do not visit them if you don't know what you're doing!
# (PL) UWAGA: Strony ponizej moga byc szkodliwe. Prosze, nie wchodz na nie, jesli nie wiesz co robisz!
#
#
#
# Podejrzane i/lub strony polaczone z innymi oszustwami
@ >>>>>>>> https://raw.githubusercontent.com/FiltersHeroes/KAD/master/sections/podejrzane_inne_oszustwa.txt => dnsmasq
address=/1vulkanvegas.com/
address=/3vulkanvegas.com/
# ...
```

Starsze listy zawierały też zerowanie:
```hosts
# ...
#@ >>>>>>>> https://raw.githubusercontent.com/FiltersHeroes/KAD/master/sections/podejrzane_inne_oszustwa.txt => dnsmasq
address=/1vulkanvegas.com/0.0.0.0
address=/3vulkanvegas.com/0.0.0.0
# ...
```

```hosts
# ...
#@ >>>>>>>> https://raw.githubusercontent.com/FiltersHeroes/KAD/master/sections/podejrzane_inne_oszustwa.txt => dnsmasq
address=/1vulkanvegas.com/0.0.0.0/
address=/3vulkanvegas.com/0.0.0.0/
# ...
```

Steven wymigał się, by inni robili konwersję:

> _https://github.com/StevenBlack/hosts#interesting-applications:_
>
>> [dnsmasq conversion script](https://gist.github.com/erlepereira/c11f4f7a3f60cd2071e79018e895fc8a#file-dnsmasq-antimalware) This GitHub gist has a short shell script (bash, will work on any 'nix) and uses wget & awk present in most distros, to fetch a specified hosts file and convert it to the format required by dnsmasq. Supports IPv4 and IPv6. Designed to be used as either a shell script, or can be dropped into /etc/cron.weekly (or wherever suits). The script is short and easily edited, also has a short document attached with notes on dnsmasq setup.

**Comment by hawkeye116477** [comment]
Raczej nie ma co dyskutować, tylko działać, tzn pewnie ja za parę dni się za to wezmę, bo raczej nie ma innych tutaj takich mistrzów wężyka :snake:  :smile:.
W ogóle jak tutaj twierdzą => https://alblue.bandlem.com/2020/05/using-dnsmasq.html i https://wiki.archlinux.org/title/dnsmasq, to można by się obyć bez dopisywania subdomen, czyli byłaby to nieco chudsza lista. Oprócz tego można zrobić ułomne regexy z samymi gwiazdkami, tzw wildcard i jakieś wykluczenia, czyli raczej nie jest najgorzej.

**Reply by hawkeye116477** [reply]
Zrobione =>https://github.com/FiltersHeroes/ScriptsPlayground/commit/161bb42ac687ad238e9a7255d1ce4cad05194345. Tylko nie ma tego zapisu na razie, bo to już zdaje się więcej roboty i mam pewne wątpliwości: 
```
# blocks google.com and all subdomains except mail.google.com.
address=/google.com/
server=/mail.google.com/#
```
Okazuje się też, że są też zwyczajne regexy jako fork => https://github.com/lixingcong/dnsmasq-regex. Nie wiem czy warto coś z tym zdziałać czy nie, pewnie trzeba byłoby przetestować, ale może po prostu najlepiej byłoby dorobić konwersję sekcji `hole_przekrety.txt` do tego formatu, o ile tak samo to działa, tylko początek i koniec ma dodatki.

**Reply by hawkeye116477** [reply]
https://raw.githubusercontent.com/FiltersHeroes/KADhosts/master/KADdnsmasq.txt
